src/components/article-body/body-components/codeblock.tsx

```diff
@@ -10,15 +12,32 @@ export const CodeBlock = ({ lang, value }: { lang: string; value: string }) => {
     const numberOfLines = getNumberOfLines(value);
     const height = getCodeHeight(numberOfLines);
     const language = getFloraLanguage(lang);
+
+    // This is a workaround to a bug where a long line of code will be cut off at the end.
+    // This seems to be a CodeMirror issue, but is especially apparent here without line wrapping.
+    // Basically this will force the component to rerender after 1 second, which allows it to recalculate the line length.
+    // https://github.com/codemirror/codemirror5/issues/5639 describes a similar issue.
+
+    const [, setReady] = useState(false);
+    useEffect(() => {
+        setTimeout(() => {
+            setReady(true);
+        }, 1000);
+    }, []);
+
+    // Need this because Safari/mobile browsers don't normally render the border radius on scroll containers without this.
+    // https://stackoverflow.com/a/58283449
+    const styling: ThemeUICSSObject = {
+        '.CodeMirror-simplescroll': { isolation: 'isolate' },
+    };
     return (
-        <div
-            // Need this because Safari/mobile browsers don't normally render the border radius on scroll containers without this.
-            // https://stackoverflow.com/a/58283449
-            sx={{
-                '.CodeMirror-simplescroll': { isolation: 'isolate' },
-            }}
-        >
-            <CodeSnippet height={height} language={language} code={value} />
+        <div sx={styling}>
```

Comment:
i touched this in the refactor PR so just a versioning mismatch here, but would you mind switching it back? just so we can avoid creating unnecessary variables

Reply:
Ah sorry about that, got confused with the merge conflict, I'll set it back to how you had it, a lot simpler that way.